specification/app/Microsoft.App.DynamicSessions/routes.tsp

```diff
@@ -0,0 +1,88 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "./main.tsp";
+import "./models/code-execution.tsp";
+import "./models/code-execution-file.tsp";
+import "./models/common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+namespace Microsoft.App.DynamicSessions;
+
+alias SessionTraits = NoRepeatableRequests &
+  NoConditionalRequests &
+  NoClientRequestId &
+  QueryParametersTrait<SessionIdentifier>;
+
+alias SessionOperations = Azure.Core.ResourceOperations<SessionTraits>;
+
+interface CodeExecution {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" ""
+  @doc("Execute code in a session.")
+  @route("/execute:executeCode")
+  executeCode is Foundations.Operation<
```

Comment:
Use LongRunningResourceAction or LongRunningResourceCollectionAction or at least LongRunningRpcOperation.

You see the warning of "use-standard-operations". And suppress without a reason is a bad practice.

Reply:
I was trying to use different options, but for this case, code execute is not really a resource, so LongRunningResourceAction and LongRunningResourceCollectionAction might not fit. And LongRunningRpcOperation only allow 202 but not 200 response status code, so it seems also not fit.

I will add suppress reasons but just would like to make sure is there any other suggestions that we could try for this scenario? Thanks.

Reply:
The LRO should return 202, not 200.
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#lro-existing-resource

Reply:
If `executionType: synchronous` then it will return 200 OK response. otherwise, 202 with location header
@weidongxu-microsoft

Reply:
Then this would be a very atypical LRO. You are taxing REST API user/customer as well as SDK dev to support a one-off LRO pattern.

Does this `executionType: synchronous` adding enough value to customer to warrant such atypical handling?
How long the REST API could take, if `executionType: synchronous`? Yon can see the guide [requires synchronous API be completed within 1 sec](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#long-running-operations--jobs).

Reply:
Looking at your Execution, it is still resource like. Please see [playground](https://azure.github.io/typespec-azure/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3JlIjsKCnVzaW5nIFR5cGVTcGVjLkh0dHA70BVSZXN00RVWyXLIG0HEWS5Db3Jl0hIuVHJhaXRzOwoKQHVzZUF1dGgoCiAgQXBpS2V5xA48xgtMb2NhdGlvbi5oZWFkZXIsICJhcGkta2V5Ij4gfCBPxCoyxS9bCiAgICB7xQYgIOQA3DrHH0Zsb3fkAKUuaW1wbGljaXQsxyVhdXRob3JpesVhVXJsOiAi5AFoczovL2xvZ2luLmNvbnRvc28uY29tL2NvbW1vbi9vxDUyL3YyLjAvyEBlIshSc2NvcGVzOiBbyUl3aWRnZXTNSi5kZWZhdWx0Il3GN30KICBdPgopCkBzZXJ2aWNlKOQAxnRpdGxlOiAiQ8Y6IFfFSSBNYW5hZ2VyIiwKfccxZXLkATkie2VuZHBvaW50fcd05QCV0D9BUElzxRnEYyAgLyoqIApTdXDkAh5lZMglU%2BYAh3MgyFNzIChwcm90b2NvbCBhbmQgaG9zdG5hbWUsIGZvciBleGFtcGxlOgrpAPRlc3R1cy5hcGku8gD%2FKS4KICovxX7IXzogc3RyaW5n5ACYfeQBCOcCvGVkKOcAvy7mAJrnAP3oAmpzKQrkAIZzcGFjZSDVKjsKCuQA41RoySDvAUgg5wFzIMdzLuQAnGVudW0gyGTlASrERccRIDIwMjItMDgtMzHELCAg5AK%2FRGVwZW5kZW5jeSjrAuLIQy52MV8wX1ByZXZpZXdfMikKICBgyUYwYCwKfQoKLy8gTW9kZWxzIC%2FTAQpAcmVzb3VyY2UoImV4ZWN15AKycyIpCm3ENCBTZXPEaUNvZGVFyB7lALhAdmlzaWJpbGl0eSgicmVhZCLEfUBkb2MoIsc1IGNvZGUgyVUgaWQuxiVrZXnLbUnGO2lk6AGtOwrJSuQBX2lkZW50aWZpZXIgb2YgdGhlIHPGYMZOyh%2FXQMlp5QO7yETvAJxyZXF1ZXN0xlPJF%2BQD3NdWc3RhdHVz105vcGVy5QP7LCBpbmRpY2F0ZXMgd2hldGjkAhh1Y2NlZWRlZCBvciBub8d0xlA6IEZvdW5kxTxzLk%2FIR1N0YXRl8ADXcnJv5wESaXPxAWtmIGZhaWxl5wFyxSw%2Fzl1FxBRSZXNwb25z0FxyZXN1bHTYXekA3sZdxi0%2FOvUCLFLFHTvkAojWX%2FABPMRU%2BgJ7xk%2FmAoHJT3N0YW5kYXJkIG91dHB121jkAWZkb3XkAKz4AdTGSesBXNlIZXLkAVLLSCNzdXBwcmVzc%2F8HDW9yZS9uby11bmtub3duIiAiIvkBfNF95QRw6gLHaXMgZmllbGQgaXMgc2FtZSBhc8U2yCJhY3R1YWzdUyBhZnRlciBiZeQHREpzb27kBLhpYWxpeukCPe4BfT86IOcAu%2FoDaGlt7gNoaW4gbWlsbGlzZWNvbmRzz1hUaW1lSW5NyyE%2FOiBpbnQ2NO8COG9wxC9hbMtraWQg5ge%2F6gI46QIt6QM%2FSWRIxSTmAjjGMO8EY%2BkBUPkC58kPSWTrAfjpAKDoBPTrBK5xdWVyeSBwYXJhbWV07ACixyxJyivuAtF1c2VyLWFzc2lnbmVk%2FwUAIEDFZ8QJbWluTGVuZ3RoKDHFGW1heMcQMjU2xRJwYXR0ZXJuKCJeW0EtWmEtejAtOS1fXXsxLDI1Nn0k%2BQVQfQoKYWxpYXPqAVzmCWMgPSBRxH5Q6ADkc8UYPPEA4T47x0HpAZTEPOsG0VLnBoXKIDzPbD47CgppbnRlcmbkB5TJHeYHRmNyZWF0ZeQDB8pBLkxvbmdSdW7kCjfIX0PFKFdpdGjnCIpQcm92aWRlZE5hbWU85Qg79AR35glO6QD0UuYGLeYCVecA8%2FoCdj4gJsU%2FICDoBW7fOck5CiAg5AD2ICDkCFfuANzoANFSZWFk6AFv7QCzxjkvLyBZb3UgY2FuIOYCey9maWxlL3vkCN19L%2BQJPGVudOQCGmFjxDUoIscT5gJaxhVTZeQCw3RvcigiL8UYZ2V05AjzZW74AJVBxTj%2FAU8ge33HCMUGICDHezogYnl0ZXM76QrBPuQCgA%3D%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fdata-plane%22%5D%7D%7D) to see how to model it and use the standard operation template.